src/edu/ucsb/nceas/metacat/admin/LoginAdmin.java

```diff
@@ -39,120 +14,291 @@
+     */
+    protected void doMetacatLogin(
+        HttpServletRequest request, HttpServletResponse response) throws AdminException {
+
+        Vector<String> processingSuccess = new Vector<>();
+        Vector<String> processingErrors = new Vector<>();
+
+        try {
+            String userId = AuthUtil.authenticateUserWithCN(request);
+
+            if (AuthUtil.isAdministrator(userId, null)) {
+                AuthUtil.setAuthCookie(request, response, -1);
+            } else {
+                processingErrors.add(userId + " is not on the Administrators list. Please contact"
+                                         + " a metacat administrator if you need access.");
+            }
+        } catch (MetacatUtilException ue) {
+            String errorMessage = "Could not log in (" + ue.getMessage() + "). Please try again";
+            processingErrors.add(errorMessage);
+            logMetacat.error(errorMessage);
+        }
+
+        try {
+            if (!processingErrors.isEmpty()) {
+                RequestUtil.setRequestErrors(request, processingErrors);
+                logMetacat.debug("Processing errors found (" + processingErrors
+                        + ").  User is not logged in; going back to login start page");
+                logOutAdminUser(request, response);
+
+            } else {
+                logMetacat.debug("Admin user logged in - authenticated and authorized");
+                processingSuccess.add("User logged in as Administrator");
+                RequestUtil.clearRequestMessages(request);
+                RequestUtil.setRequestSuccess(request, processingSuccess);
+                RequestUtil.forwardRequest(request, response,
+                                           MetacatAdminServlet.PATH_ADMIN_HOMEPAGE,
+                                           null);
+            }
+        } catch (MetacatUtilException mue) {
+            AdminException adminException = new AdminException(
+                "Problem while processing login; unable to forward request: " + mue.getMessage());
+            adminException.fillInStackTrace();
+            throw adminException;
+        }
+    }
+
+    /**
+     * Put user in a logged-out state by removing userId from session, and invalidating auth cookie.
+     * Then return to login page
+     *
+     * @param request  The http request information
+     * @param response The http response to be sent back to the client
+     * @throws AdminException if unable to invalidate auth cookie or forward request
+     */
+    protected void logOutAdminUser(
+        HttpServletRequest request, HttpServletResponse response) throws AdminException {
+
+        logMetacat.debug(
+            "logOutAdminUser - logging out... " + request.getSession().getAttribute("userId"));
+
+        request.getSession().removeAttribute("userId");
+
+        addProcessingMessage(request, "Successfully logged out");
+
+        try {
+            AuthUtil.invalidateAuthCookie(request, response);
```

Comment:
This is a little confusing at first glance. According to my research, in order to write server-side code that can "remove" a client-side cookie, you have to basically set the cookie timeout value to zero - so in this case, `invalidateAuthCookie` calls `setAuthCookie` and passes `timeoutMins` = 0

Reply:
Ah, that makes sense! Now it's connecting together.

Reply:
FYI, I also [added this line earlier today](https://github.com/NCEAS/metacat/pull/1826/files/a1845b4e7337e2736902158bf79e275070ad2605..f8183ccd13b2ffb1e8bd45c730b1618189c8af42#diff-87fbc4d87f1cf97ee20f2d8f335cc503d4a44088795cd3dc3b9a20189b20034fR234), as a best practice

Reply:
Looks good! I think my confusion began because I was tunnel-visioning for an `Expires` cookie-name in `setAuthCookie`, which seems redundant now since you can immediately get the client to remove the cookie by setting `Max-Age`.